receiver/sqlqueryreceiver/db_client.go

```diff
@@ -8,9 +8,10 @@ import (
 
 	// register db drivers
 	_ "github.com/SAP/go-hdb/driver"
-	_ "github.com/denisenkom/go-mssqldb"
 	_ "github.com/go-sql-driver/mysql"
 	_ "github.com/lib/pq"
+	_ "github.com/microsoft/go-mssqldb"
+	_ "github.com/microsoft/go-mssqldb/integratedauth/krb5"
```

Comment:
Why do we need this?

Reply:
This is what the change is - Swapping the old 'denisenkom' driver for the new microsoft driver.

With regards to the github.com/microsoft/go-mssqldb/integratedauth/krb5 module, this is required to be able to use kerberos authentication mechanism.

Reply:
Why only this one? Other authentication mechanisms also have `init()` func https://github.com/microsoft/go-mssqldb/tree/main/integratedauth

Reply:
Good point.

The reason that krb5 is added is because that is actually the auth mechanism that I have used so far myself.

We may actually need the other additional ones as well!

I will investigate and come back shortly.

Reply:
This is the source:
https://github.com/microsoft/go-mssqldb/blob/c902b674869e3f177b723cf2a5b1364c14efda47/README.md?plain=1#L93-L125

Basically, the summary is:

The driver has support to perform three different "integrated auth" mechanism, depending on which platform the driver is executing on:

https://github.com/microsoft/go-mssqldb/blob/main/README.md?plain=1#L129

- On Windows, the default auth mechanism is something called `SSPI`, which is only available when building the driver for Windows https://github.com/microsoft/go-mssqldb/tree/main/integratedauth/winsspi (which basically automatically will chose either kerberos or ntlm)
- On Linux, the two options are either `ntlm` or `krb5` explicitely, with `ntlm` being the default in the driver

I do not know why the `krb5` package was designed in such way that it needs an explicit import.

There is some dicsussion on the original thread when the feature was implemented, but I can't quickly glean why this kerberos features was designed in this particular way.

https://github.com/microsoft/go-mssqldb/pull/35

Reply:
> Why only this one? Other authentication mechanisms also have `init()`

In summary. The Kerberos on Non-Windows integratedauth package is special compared to the winsspi and ntlm packages in that it is required to be imported explicitly, but I have no understanding of why the `krb5` package was designed in this way.